webknossos/webknossos/dataset/_utils/pims_czi_reader.py

```diff
@@ -0,0 +1,112 @@
+from contextlib import contextmanager
+from os import PathLike
+from pathlib import Path
+from typing import Iterator, List, Set
+
+import numpy as np
+
+try:
+    from pims import FramesSequenceND
+except ImportError as e:
+    raise RuntimeError(
+        "Cannot import pims, please install it e.g. using 'webknossos[all]'"
+    ) from e
+
+try:
+    from pylibCZIrw import czi as pyczi
+except ImportError as e:
+    raise RuntimeError(
+        "Cannot import pylibCZIrw, please install it e.g. using 'webknossos[czi]'"
+    ) from e
+
+PIXEL_TYPE_TO_DTYPE = {
+    "Gray8": "<u1",
+    "Gray16": "<u2",
+    "Gray32Float": "<f4",
+    "Bgr24": "<u1",
+    "Bgr48": "<u2",
+    "Bgr96Float": "<f4",
+    "Bgra32": "<4u1",
+    "Gray64ComplexFloat": "<F8",
+    "Bgr192ComplexFloat": "<F8",
+    "Gray32": "<i4",
+    "Gray64": "<i8",
+}
+
+
+class PimsCziReader(FramesSequenceND):
+    @classmethod
+    def class_exts(cls) -> Set[str]:
+        return {".czi"}
+
+    class_priority = 20
```

Comment:
what is this?

Reply:
This is used in pims to pick the reader with the highest priority. Default is 10, and bioformats priority (which is the only other reader supporting czi) is 2. From the [pims docs](http://soft-matter.github.io/pims/v0.6.1/custom_readers.html#plugging-into-pims-s-open-function):
> To prioritize readers, a field class_priority can optionally be given to the reader. A higher priority will be chosen over a lower priority. Default for all readers is 10.

Reply:
Also added as a comment.